cirq/contrib/noise_models/noise_models.py

```diff
@@ -0,0 +1,152 @@
+                pass
+            else:
+                raise ValueError("Moment must be all measurements "
+                                 "or all operations.")
+        else:
+            if is_measurements is None:
+                is_measurements = False
+            elif not is_measurements:
+                # Check that all are *not* measurements
+                pass
+            else:
+                raise ValueError("Moment must be all measurements "
+                                 "or all operations.")
+    return is_measurements
+
+
+class DepolarizingNoiseModel(devices.NoiseModel):
+    """Applies depolarizing noise to each qubit individually at the end of
+    every moment.
+
+    If a circuit contains measurements, they be in moments that don't
+    also contain gates.
+
+    Args:
+        depol_prob: Depolarizing probability.
+    """
+
+    def __init__(self, depol_prob: float):
+        value.validate_probability(depol_prob, 'depol prob')
+        self.qubit_noise_gate = ops.DepolarizingChannel(depol_prob)
+
+    def noisy_moment(self, moment: 'cirq.Moment',
+                     system_qubits: Sequence['cirq.Qid']):
+        if _moment_is_measurements(moment):
+            return moment
+
+        return [
+            moment,
+            ops.Moment(self.qubit_noise_gate(q) for q in system_qubits)
+        ]
+
+
+class DepolarizingWithReadoutNoiseModel(devices.NoiseModel):
+    """DepolarizingNoiseModel with probabilistic bit flips preceding
+    measurement.
+
+    This simulates readout error.
+
+    If a circuit contains measurements, they be in moments that don't
+    also contain gates.
+
+    Args:
+        depol_prob: Depolarizing probability.
+        bitflip_prob: Probability of a bit-flip during measurement.
+    """
+
+    def __init__(self, depol_prob: float, bitflip_prob: float):
+        value.validate_probability(depol_prob, 'depol prob')
+        value.validate_probability(bitflip_prob, 'bitflip prob')
+        self.qubit_noise_gate = ops.DepolarizingChannel(depol_prob)
+        self.readout_noise_gate = ops.BitFlipChannel(bitflip_prob)
+
+    def noisy_moment(self, moment: 'cirq.Moment',
+                     system_qubits: Sequence['cirq.Qid']):
+        if _moment_is_measurements(moment):
+            return [
+                ops.Moment(self.readout_noise_gate(q) for q in system_qubits),
+                moment,
+            ]
+        return [
+            moment,
+            ops.Moment(self.qubit_noise_gate(q) for q in system_qubits),
+        ]
+
+
+class DepolarizingWithDampedReadoutNoiseModel(devices.NoiseModel):
+    """DepolarizingWithReadoutNoiseModel with T1 decay preceding
+    measurement.
+
+    This simulates asymmetric readout error. The noise is structured
+    so the T1 decay is applied, then the readout bitflip, then measurement.
+
+    If a circuit contains measurements, they be in moments that don't
+    also contain gates.
+
+    Args:
```

Comment:
Does listing the arguments here instead of on `__init__` work? Will they show up in pycharm's autocomplete?

Reply:
I've done some investigating. Pep257 says to put constructor docs in `__init__` but numpy (and presumably others) put it all in the class.

Pycharm autocomplete pulls the names from the function signature and not the docstring, so it still works. Pycharm "quick documentation" *does* correctly parse the Args documentation from class docstrings. 

I can put this on `__init__` if that's what the rest of cirq does

Reply:
Yeah, that's what the rest of cirq happens to do.